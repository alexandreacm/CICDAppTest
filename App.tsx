import { SafeAreaView } from 'react-native-safe-area-context';
import HomeScreen from './src/screens/HomeScreen';
import StatusBar from './src/components/StatusBar';
import { useColorScheme } from 'react-native';

function App() {
  const isDarkMode = useColorScheme() === 'dark';
  return (
    // eslint-disable-next-line react-native/no-inline-styles
    //NEw COMMENT
    //I am here....
    //other thing bla bla bla
    //new test to push the content to main branch
    //feature-4
    <SafeAreaView style={{ flex: 1 }}>
      <StatusBar isDarkMode={isDarkMode} />
      <HomeScreen isDarkMode={isDarkMode} />
    </SafeAreaView>
  );
}

export default App;
