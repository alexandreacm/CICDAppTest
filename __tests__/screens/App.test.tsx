/**
 * @format
 */

import React from 'react';
// import ReactTestRenderer from 'react-test-renderer';
import App from '../../App';
import { render, screen } from '@testing-library/react-native';

describe('<App />', () => {
  beforeEach(() => {
    jest.resetModules();
  });
  test('Should renders correctly',  () => {
    render(<App />)
  });

  test('Should renders snapshot',  () => {
    const { toJSON } = render(<App />);

    expect(toJSON()).toMatchSnapshot();
  });

   test('Should render Welcome text ',  () => {
    render(<App />);

    screen.getByText('Welcome to React Native!');
  });
});
