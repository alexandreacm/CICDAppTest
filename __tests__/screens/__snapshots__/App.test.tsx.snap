// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Should renders snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        {
          "backgroundColor": "#FFFFFF",
        },
      ]
    }
  >
    <View>
      <Text
        style={
          [
            {
              "fontSize": 20,
              "fontWeight": "bold",
            },
            {
              "color": "#000000",
            },
          ]
        }
      >
        Welcome to React Native!
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;
